packages/vaadin-icon/src/vaadin-icon.js

```diff
@@ -105,12 +105,15 @@ class IconElement extends ThemableMixin(ElementMixin(PolymerElement)) {
   __iconChanged(icon) {
     if (icon) {
       const parts = icon.split(':');
-      const iconName = getIconId(parts.pop());
-      const iconsetName = parts.pop() || DEFAULT_ICONSET;
+      const iconsetName = parts[0] || DEFAULT_ICONSET;
       const iconset = IconsetElement.getIconset(iconsetName);
-      this.svg = iconset.applyIcon(iconName);
+      const { svg, size } = iconset.applyIcon(icon);
+      if (size !== this.size) {
+        this.size = size;
```

Comment:
What happens when both an icon and an icon set have custom `size` values, and these sizes differ? Should this case be documented?

```html
<vaadin-iconset name="acme" size="32">...</vaadin-iconset>
<vaadin-icon icon="acme:up" size="16"/>
```

Reply:
Updated docs to mention that `vaadin-iconset` overrides the properties set on the icon.